heroic-core/src/main/java/com/spotify/heroic/servlet/MandatoryClientIdFilter.java

```diff
@@ -62,27 +66,28 @@ public void init(FilterConfig filterConfig) throws ServletException { /* intenti
      * to the client.
      */
     @Override
-    public void doFilter(
-        ServletRequest request, ServletResponse response, FilterChain chain
+    public ErrorMessage doFilterImpl(
+            HttpServletRequest request, HttpServletResponse response, FilterChain chain
     ) throws IOException, ServletException {
-        if (passesFilter(request)) {
-            chain.doFilter(request, response);
-        } else {
-            var httpResponse = HttpServletResponse.class.cast(response);
-            httpResponse.setStatus(Status.BAD_REQUEST.getStatusCode());
-        }
+
+        final var info =
+                new MandatoryClientIdErrorMessage("Anonymous requests are not permitted",
+                        Status.SERVICE_UNAVAILABLE);
```

Comment:
Is this supposed to be `Status.BAD_REQUEST`?

Reply:
Yes! That explains the 503 prolly.